### DISCUSSION TITLE
how to add platform, os and python version to wheel's filename?

### DISCUSSION THREAD

**Discussion by zillionare** [python-poetry/poetry#4909] [Q&A]
My toml file is:

```
[tool.poetry]
name = "zillionare-omega"
packages = [
  {include = "omega"}
]


classifiers = [
    "Development Status :: 3 - Alpha",
    "Intended Audience :: Developers",
    "License :: OSI Approved :: MIT License",
    "Natural Language :: English",
    "Programming Language :: Python :: 3.9",
    "Operating System :: POSIX :: Linux"
]

include = [
...
]

version = "1.1.1"

[tool.poetry.dependencies]
python=">=3.9,<3.10"
...

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

after run `poetry build`, the name of the wheel is:
zillionare_omega-1.1.1-py3-none-any.whl

what I want is zillionare_omega_1.1.1-py39-x86-many_linux.whl

How can I do that?

**Comment by clintonroy** [comment]
Basically at the moment, if you don't use a build script, poetry will use a platform of "any".

If you use a build script, poetry will grab the tightest platform of the running python.

Both of these can be considered less than ideal. If you really want to modify the platform (for example, you're using linux specific /proc files)  then I would suggest that poetry isn't an ideal choice for now.

**Reply by zillionare** [reply]
Thanks for help!

I'm asking because I was stuck in releasing two packages for apple m1.

These two packages will only support Ubuntu, apple m1. And in package one, it depends on bottleneck, scipy and etc, these all requires specified packages for apple m1. and package tow will depends on package one.

I know I should use markers to specify different binary for different platform. However it doesn't goes smoothly. So I have to decide keep one branch is dedicated to apple m1 for each project, and release apple m1 packages from this branch.

then in package one's pyporject.toml, I can just use:
```
bottlenect = "https://package_specific_for_apple_m1"
```

and in package two's toml, I can use:
```
one = "https://package_one_m1
```

so I'll need a specific name for package one.

After read your and other replies, I understand now I can put an empty build script and add "build = " to toml file.

By the way, would you please inform me what's the best practice in my case?

**Reply by clintonroy** [reply]
tool.poetry.build support is undocumented and subject to breaking changes, it's not really supported at all, so there's no hope of having best practices with it right now.
